src/utils/expansion.rs

```diff
@@ -0,0 +1,113 @@
+use anyhow::{anyhow, bail, Context, Error};
+use std::env;
+use std::fs::read_to_string;
+use std::path::Path;
+
+use shellexpand::env_with_context;
+
+use crate::RoverResult;
+
+/// Implements router-config-style
+/// [variable expansion](https://www.apollographql.com/docs/router/configuration/overview/#variable-expansion)
+/// for use in Rover configs (like `supergraph.yaml`).
+pub(crate) fn expand(value: &str) -> RoverResult<String> {
+    env_with_context(value, context)
+        .map_err(|e| anyhow!(e).context("While expanding variables").into())
+        .map(|cow| cow.into_owned())
+}
+
+fn context(key: &str) -> Result<Option<String>, Error> {
+    if let Some(env_var_key) = key.strip_prefix("env.") {
+        env::var(env_var_key).map(Some).with_context(|| {
+            format!(
+                "While reading env var {} for variable expansion",
+                env_var_key
+            )
+        })
+    } else if let Some(file_name) = key.strip_prefix("file.") {
+        if !Path::new(key).exists() {
+            Ok(None)
+        } else {
+            read_to_string(file_name).map(Some).with_context(|| {
```

Comment:
would you mind using `rover_std::Fs::read_file` here instead? it has a couple wrappers around it for error handling and logging that I would like to keep consistent